### PR TITLE
added icons and handling of flagstatus in the ui

### PR DESCRIPTION
**Pull request by Sepja**

### CODE REVIEW

File: src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java [hunk 1]
```diff
@@ -1,59 +1,66 @@
...
+    private final int WIDTH_PX = 25;
+    private final int HEIGHT_PX = 25;
+    private final IconProvider iconProvider;
 
-        if(cell.isCovered()){
-            label.setId("covered");
-            label.setOnMouseClicked(mouseEventHandler);
+    public MineFieldCreator(IconProvider iconProvider){
+        this.iconProvider = iconProvider;
+
```
Comment (pierre12) [inline src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java]: remove empty line

File: src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java [hunk 2]
```diff
@@ -1,59 +1,66 @@
...
+    }
+
+    private Label createCoveredField(Cell cell, EventHandler<MouseEvent> mouseEventHandler) {
+        Label label = createStandardLabel();
+        label.setId("covered");
+        label.setOnMouseClicked(mouseEventHandler);
+        if (cell.getFlagStatus() == FlagStatus.MARKED_AS_MINE) {
+            label.setGraphic(this.iconProvider.createMineMarkerImage(WIDTH_PX, HEIGHT_PX));
+        } else if (cell.getFlagStatus() == FlagStatus.MARKED_AS_UNKNOWN) {
+            label.setGraphic(this.iconProvider.createQuestionMarkMarkerImage(  WIDTH_PX, HEIGHT_PX));
```
Comment (pierre12) [inline src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java]: (  WIDTH_PX, HEIGHT_PX) <= remove space after opening bracket

File: src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java [hunk 3]
```diff
@@ -1,59 +1,66 @@
...
+            if (cell.getSurroundingMines() > 0) {
                 label.setText(String.valueOf(cell.getSurroundingMines()));
             }
         }
 
-        if(isUncoveredMine(cell )){
+        if (cell.isMine()) {
             label.setId("bomb");
-            label.setGraphic(createImage("/icons/Bomb.png",25,25));
+            label.setGraphic(this.iconProvider.createMineImage( WIDTH_PX, HEIGHT_PX));
```
Comment (pierre12) [inline src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java]: ( WIDTH_PX, HEIGHT_PX) <= remove space after opening bracket

File: src/test/java/com/labuhn/minesweeper/ui/controller/MineCellCreatorTest.java [hunk 4]
```diff
@@ -77,14 +81,17 @@ public void createCoveredFieldWithSurroundingMines(){
...
+        Mockito.when(iconProvider.createMineImage(25,25)).thenReturn(expectedImageView);
         Label cell = mineFieldCreator.createCell(new Cell(false, true, 4), null);
 
         assertThat(cell.getMinHeight()).isEqualTo(25);
         assertThat(cell.getMaxHeight()).isEqualTo(25);
         assertThat(cell.getMinWidth()).isEqualTo(25);
         assertThat(cell.getMaxHeight()).isEqualTo(25);
         assertThat(cell.getId()).isEqualTo("bomb");
-        assertThat(cell.getGraphic().getClass()).isEqualTo(ImageView.class);
+        Mockito.verify(iconProvider,Mockito.times(1)).createMineImage(25,25);
```
Comment (pierre12) [inline src/test/java/com/labuhn/minesweeper/ui/controller/MineCellCreatorTest.java]: try static import of Mockito. Makes statement more readable

**Review by pierre12 — CHANGES_REQUESTED** [review]

File: src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java (hunk 1 above)
Comment (Sepja) [reply src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java]: removed

File: src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java (hunk 2 above)
Comment (Sepja) [reply src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java]: removed

File: src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java (hunk 3 above)
Comment (Sepja) [reply src/main/java/com/labuhn/minesweeper/ui/controller/MineFieldCreator.java]: removed

File: src/test/java/com/labuhn/minesweeper/ui/controller/MineCellCreatorTest.java (hunk 4 above)
Comment (Sepja) [reply src/test/java/com/labuhn/minesweeper/ui/controller/MineCellCreatorTest.java]: added static import

**Review by pierre12 — APPROVED** [review]

**Pull request merged**